tools/privacyidea-migrate-linotp.py

```diff
@@ -1,84 +1,161 @@
 
-"tokens" should always be True.
-If "assignments" is False, the imported tokens will be unassigned!
-
-"""
-MIGRATE = {"tokens": True,
-           "tokeninfo": True,
-           "assignments": True}
+MIGRATE defines which elements should be migrated, tokens, tokeninfo and assignment?
 
-"""
-This assigns the token to the new resolver
+The ASSIGNMENTS section defines, who the token-assignments should be migrated.
 
-The "resolver" key maps LinOTP-resolvers to privacyIDEA-Resolvers. This value is
-rewritten in the token table.
+The "resolver" key maps LinOTP-resolvers to privacyIDEA-Resolvers. 
+The key is the LinOTP resolver and the value is the new privacyIDEA resolver.
 
-The "realm" key puts tokens in the privacyIDEA-Resolver into this privacyIDEA realm.
+The "realm" key puts tokens in the specified privacyIDEA resolver (key)
+into the privacyIDEA realm (value). Note: There is no logic checking done,
+so take care that the resolver is really located in the specified realm.
 
 In this example the resolvername "lokal" from LinOTP gets imported to
 privacyIDEA with the resolver "PIResolver" in realm "pirealm".
 
-Tokens, that were not assigned to a user will be assigned to the realms "realm1" and "realm2"
-in privacyIDEA.
-"""
-ASSIGNMENTS = {
-    "resolver": {"lokal": "PIResolver"},
-    "realm": {"PIResolver": "pirealm"},
-    "unassigned_tokens": ["realm1", "realm2"]
-}
+Tokens, that were not assigned to a user will be assigned to the realms specified
+in the list "unassigned_tokens".
 
+The NEW_TOKENINFO section can set any arbitrary tokeninfo for the 
+migrated tokens.
 """
-This option splits INSERT statements into chunks of a given size, which is useful
-if the number of migrated tokens is so large that it cannot fit in one INSERT
-statement.
-"""
-INSERT_CHUNK_SIZE = 10000
-
-# Do not change anything after this line
-# ============================================================================
-#
-
-
-def migrate():
+from sqlalchemy import create_engine
+from sqlalchemy.orm import sessionmaker
+from sqlalchemy.sql import select
+from sqlalchemy.schema import ForeignKey
+from sqlalchemy import (Table, MetaData, Column, Integer, Unicode, Boolean,
+                        UnicodeText)
+import sys
+import json
+import getopt
+import binascii
+import os
+import re
+from privacyidea.lib.crypto import aes_cbc_decrypt, aes_cbc_encrypt
+from privacyidea.lib.utils import hexlify_and_unicode
+
+
+EXAMPLE_CONFIG_FILE = """{
+    "SQL": {
+        "LINOTP_URI": "mysql+pymysql://linotp:linotp@localhost/linotp2",
+        "PRIVACYIDEA_URI": "mysql+pymysql://pi:QEK_k_f3SGbu@localhost/pi?charset=utf8mb4",
+        "INSERT_CHUNK_SIZE": 10000
+    },
+    "MIGRATE": {
+        "tokens": true,
+        "tokeninfo": true,
+        "assignments": false
+    },
+    "ASSIGNMENTS": {
+        "resolver": {"local": "localpw"},
+        "realm": {"localpw": "neuerrealm"},
+        "unassigned_tokens": ["testfoo"],
+        "convert_endian": true
```

Comment:
Please describe `convert_endian` option.

Reply:
Oh, I did not do it?
Will do.

Reply:
Done in ad247324